org.eclipse.swtchart/src/org/eclipse/swtchart/internal/series/MultiLevelPie.java

```diff
@@ -0,0 +1,258 @@
+ * Copyright (c) 2020 SWTChart project.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ * Himanshu Balasamanta - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swtchart.internal.series;
+
+import java.util.HashMap;
+import java.util.Set;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.GC;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swtchart.Chart;
+import org.eclipse.swtchart.IAxis;
+import org.eclipse.swtchart.IMultiLevelPie;
+import org.eclipse.swtchart.Range;
+import org.eclipse.swtchart.internal.axis.Axis;
+import org.eclipse.swtchart.internal.compress.Compress;
+import org.eclipse.swtchart.internal.compress.CompressMultiLevelPie;
+import org.eclipse.swtchart.model.IdNodeDataModel;
+import org.eclipse.swtchart.model.Node;
+
+public class MultiLevelPie extends Series implements IMultiLevelPie {
+
+	private Chart chart;
+	private IdNodeDataModel model;
+	private Node rootNode;
+	private int maxTreeDepth;
+
+	protected MultiLevelPie(Chart chart, String id) {
+
+		super(chart, id);
+		this.chart = chart;
+		type = SeriesType.MULTI_LEVEL_PIE;
+		initialise();
+		model = new IdNodeDataModel(id, this);
+		rootNode = model.getRootNode();
+		compressor = new CompressMultiLevelPie(model);
+	}
+
+	@Override
+	public Compress getCompressor() {
+
+		return (Compress)compressor;
+	}
+
+	@Override
+	public Node getRootNode() {
+
+		return rootNode;
+	}
+
+	@Override
+	public Node getNodeById(String id) {
+
+		return model.getTree().get(id);
+	}
+
+	@Override
+	public void addSeries(String[] labels, double[] values) {
+
+		int length = labels.length;
+		if(values.length != length) {
+			// throw error
+		}
+		for(int i = 0; i != length; i++) {
+			Node node = new Node(labels[i], values[i], rootNode);
+		}
+		update();
+	}
+
+	/**
+	 * This returns the orignal series that we provided as input.
+	 * This is added here so as to provide the functionalities of the simple pie chart
+	 * 
+	 * @return
+	 */
+	@Override
+	public HashMap<String, Node> getSeries() {
+
+		return rootNode.getChildren();
+	}
+
+	@Override
+	public void addNode(String id, double val) {
+
+		rootNode.addChild(id, val);
+	}
+
+	@Override
+	public String[] getLabels() {
+
+		Set<String> nodeIds = model.getTree().keySet();
+		String[] ids = new String[nodeIds.size()];
+		System.arraycopy(nodeIds.toArray(), 0, ids, 0, nodeIds.size());
+		return ids;
+	}
+
+	@Override
+	public Color[] getColors() {
+
+		return null;
+	}
+
+	@Override
+	public Range getAdjustedRange(Axis axis, int length) {
+
+		return null;
+	}
+
+	@Override
+	protected void setCompressor() {
+
+		((CompressMultiLevelPie)compressor).update();
```

Comment:
Kindly comment out this line and run the "MultiLevelPieChartExample" class to run without errors